src/GrpcProtos/capture.proto

```diff
@@ -599,6 +599,28 @@ message ThreadStateSlice {
   // Otherwise, they should be 0 to minimize bandwidth usage.
   uint32 wakeup_tid = 7;
   uint32 wakeup_pid = 8;
+
+  // This is the callstack of the tracepoint that changed the state
+  // of the thread and caused this slice to happen. If the thread
+  // state is runnable then this callstack is coming from wakeup_tid,
+  // otherwise it's coming from the same thread.
+  uint64 triggering_callstack_id = 10;
```

Comment:
I this we should call this `switch_out_or_wakeup_callstack_id`. This is to clarify that there are two clearly separate cases for this callstack, as described above. And `wakeup` makes the association with `wakeup_tid`, `wakeup_reason`, etc.

Reply:
Done.